internal/service/autoscaling/policy.go

```diff
@@ -793,3 +1163,81 @@ func flattenPredefinedMetricPairSpecification(predefinedMetricPairSpecification
 	predefinedMetricPairSpecificationFlat["resource_label"] = aws.StringValue(predefinedMetricPairSpecification.ResourceLabel)
 	return []map[string]interface{}{predefinedMetricPairSpecificationFlat}
 }
+
+func flattenCustomizedScalingMetricSpecification(customizedScalingMetricSpecification *autoscaling.PredictiveScalingCustomizedScalingMetric) []map[string]interface{} {
+	customizedScalingMetricSpecificationFlat := map[string]interface{}{}
+	if customizedScalingMetricSpecification == nil {
+		return []map[string]interface{}{customizedScalingMetricSpecificationFlat}
+	}
+	customizedScalingMetricSpecificationFlat["metric_data_queries"] = flattenMetricDataQueries(customizedScalingMetricSpecification.MetricDataQueries)
+	return []map[string]interface{}{customizedScalingMetricSpecificationFlat}
+}
+
+func flattenCustomizedLoadMetricSpecification(customizedLoadMetricSpecification *autoscaling.PredictiveScalingCustomizedLoadMetric) []map[string]interface{} {
+	customizedLoadMetricSpecificationFlat := map[string]interface{}{}
+	if customizedLoadMetricSpecification == nil {
+		return []map[string]interface{}{customizedLoadMetricSpecificationFlat}
+	}
+	customizedLoadMetricSpecificationFlat["metric_data_queries"] = flattenMetricDataQueries(customizedLoadMetricSpecification.MetricDataQueries)
+	return []map[string]interface{}{customizedLoadMetricSpecificationFlat}
+}
+
+func flattenCustomizedCapacityMetricSpecification(customizedCapacityMetricSpecification *autoscaling.PredictiveScalingCustomizedCapacityMetric) []map[string]interface{} {
+	customizedCapacityMetricSpecificationFlat := map[string]interface{}{}
+	if customizedCapacityMetricSpecification == nil {
+		return []map[string]interface{}{customizedCapacityMetricSpecificationFlat}
+	}
+	customizedCapacityMetricSpecificationFlat["metric_data_queries"] = flattenMetricDataQueries(customizedCapacityMetricSpecification.MetricDataQueries)
+
+	return []map[string]interface{}{customizedCapacityMetricSpecificationFlat}
+}
+
+func flattenMetricDataQueries(metricDataQueries []*autoscaling.MetricDataQuery) []interface{} {
+	metricDataQueriesFlat := map[string]interface{}{}
+	if metricDataQueriesFlat == nil {
+		return []interface{}{}
+	}
+
+	metricDataQueriesSpec := make([]interface{}, len(metricDataQueries))
+	for i := range metricDataQueriesSpec {
+		metricDataQuery := map[string]interface{}{}
+		rawMetricDataQuery := metricDataQueries[i]
+		metricDataQuery["id"] = aws.StringValue(rawMetricDataQuery.Id)
+		if rawMetricDataQuery.Expression != nil {
+			metricDataQuery["expression"] = aws.StringValue(rawMetricDataQuery.Expression)
+		}
+		if rawMetricDataQuery.MetricStat != nil {
+			metricStatSpec := map[string]interface{}{}
+			rawMetricStat := rawMetricDataQuery.MetricStat
+			metricStatSpec["stat"] = aws.StringValue(rawMetricStat.Stat)
+			if rawMetricStat.Unit != nil {
```

Comment:
Small style nit: Following the same alphabetical order convention for the optional attributes, perhaps `dimensions` can come before `stat`. Since there are also multiple nested attributes adding some comments would make the code more readable

Reply:
`dimensions` and `stat` are in fact from different structure, but I will refactor fields from same structure with  alphabetical order.